app/src/main/java/org/oppia/android/app/mydownloads/downloads/DownloadsFragmentPresenter.kt

```diff
@@ -0,0 +1,205 @@
+    downloadsViewModel.setProfileId(internalProfileId)
+
+    binding.downloadsRecyclerView.apply {
+      downloadsRecyclerViewAdapter = createRecyclerViewAdapter()
+      adapter = downloadsRecyclerViewAdapter
+    }
+    return binding.root
+  }
+
+  private fun createRecyclerViewAdapter(): BindableAdapter<DownloadsItemViewModel> {
+    return BindableAdapter.MultiTypeBuilder
+      .newBuilder<DownloadsItemViewModel, ViewType> { downloadsItemViewModel ->
+        when (downloadsItemViewModel) {
+          is DownloadsSortByViewModel -> ViewType.VIEW_TYPE_SORT_BY
+          is DownloadsTopicViewModel -> ViewType.VIEW_TYPE_TOPIC
+          else -> throw IllegalArgumentException(
+            "Encountered unexpected view model: $downloadsItemViewModel"
+          )
+        }
+      }
+      .registerViewDataBinder(
+        viewType = ViewType.VIEW_TYPE_SORT_BY,
+        inflateDataBinding = DownloadsSortbyBinding::inflate,
+        setViewModel = this::bindDownloadsSortBy,
+        transformViewModel = { it as DownloadsSortByViewModel }
+      )
+      .registerViewDataBinder(
+        viewType = ViewType.VIEW_TYPE_TOPIC,
+        inflateDataBinding = DownloadsTopicCardBinding::inflate,
+        setViewModel = this::bindDownloadsTopicCard,
+        transformViewModel = { it as DownloadsTopicViewModel }
+      )
+      .build()
+  }
+
+  private enum class ViewType {
+    VIEW_TYPE_SORT_BY,
+    VIEW_TYPE_TOPIC
+  }
+
+  private fun bindDownloadsSortBy(
+    binding: DownloadsSortbyBinding,
+    viewModel: DownloadsSortByViewModel
+  ) {
+    binding.viewModel = viewModel
+    val adapter = ArrayAdapter(
+      fragment.requireContext(),
+      R.layout.downloads_sortby_menu,
+      SortByItems.values()
+    )
+    // setting input type to zero makes AutoCompleteTextView no editable
+    binding.sortByMenu.setInputType(0)
+    binding.sortByMenu.setText(adapter.getItem(previousSortTypeIndex ?: 0).toString(), false)
+    binding.sortByMenu.setAdapter(adapter)
+
+    // TODO(#552): orientation change, keep list sorted as per previousSortTypeIndex
+
+    binding.sortByMenu.setOnItemClickListener { parent, view, position, id ->
+      if (previousSortTypeIndex != position) {
+        when (parent.getItemAtPosition(position)) {
+          SortByItems.NEWEST -> {
+            // TODO(#552)
+          }
+          SortByItems.ALPHABETICAL -> {
+            sortTopicAlphabetically()
+          }
+          SortByItems.DOWNLOAD_SIZE -> {
+            // TODO(#552)
+          }
+        }
+        previousSortTypeIndex = position
+        sortByListIndexListener.onSortByItemClicked(previousSortTypeIndex)
+        binding.sortByMenu.setText(adapter.getItem(position).toString(), false)
+      }
+    }
+  }
+
+  private fun sortTopicAlphabetically() {
+    val sortedTopicNameList = mutableListOf<String>()
+    val downloadsItemViewModelList = downloadsViewModel.downloadsViewModelLiveData.getValue()
```

Comment:
`val downloadsItemViewModelList = downloadsViewModel.downloadsViewModelLiveData.value`

Reply:
done

Reply:
resolving as this is nit fix